### ISSUE TITLE
Unwind error in game

### ISSUE THREAD

**Issue by paulo-coutinho** [emscripten-core/emscripten#16071]
Please include the following in your bug report:

**Version of emscripten/emsdk:**
3.1

My simple game is throwing:
```
Uncaught (in promise) unwind
```

There is a code in generate .js file:

```
if (simulateInfiniteLoop) {
    throw "unwind"
}
```

![image](https://user-images.githubusercontent.com/395096/150266927-58e1cf58-35b0-4cbb-80d9-a15f72e43ea8.png)

Demo:
https://nativium.s3.amazonaws.com/game/1.0.0/index.html

This only happen when i start the game from function `Game.Core.Initialize(640, 640, 1);`. If i let it from `main` function, no error happen.

What im doing wrong?

Thanks.

**Comment by kripken** [comment]
That exception is thrown to return to the main event loop. If it were not thrown then code after the call to `emscripten_main_loop` would continue, which shouldn't. So this is expected, if you told `emscripten_main_loop` or another function that you want to return to the main loop, and you can ignore it.

**Comment by paulo-coutinho** [comment]
Hi @kripken,

There is some way to start the main loop from a function (not `main`) without throw this error?

My main.cpp:

```
#ifdef __cplusplus
extern "C"
{
#endif

    EMSCRIPTEN_KEEPALIVE void Nativium_Game_Initialize(const int32_t width, const int32_t height, const int32_t mode)
    {
        ApplicationCore::shared()->initialize(width, height, mode);
    }

#ifdef __cplusplus
}
#endif

int main(int argc, char **argv)
{
    ApplicationCore::shared()->initialize(800, 640, 1); // here is the game start and main loop
    return EXIT_SUCCESS;
}
```

It only throw this error only when i delete my `int main` method and start the game from the function `Nativium_Game_Initialize`. I don't understand why, since the main loop was not touched, but started from different ways.

**Comment by sbc100** [comment]
I believe all you need to do is pass 0/false as the third argument to `emscripten_set_main_loop` (`simulate_infinite_loop`).   If you don't need to use an exception to break out of the main function then there is no need to pass 1/true here.

**Comment by paulo-coutinho** [comment]
Hi @sbc100,

It can have any performance issue?

**Comment by sbc100** [comment]
No.. all that flag does is select whether or not to throw the unwind exception.
